### PR TITLE
Set seeds

### PR DESCRIPTION
**Pull request by mosheka**
1.  Move from cleaning queues task on test begining to marking messages w/ phase tags, and avoiding old/other phases
2. Moving to logging package
3. Create dedicated config files and modify the seeds on them
4. Support multiple nodes on both logging and configuration
5. Encapsulate docker related integration
6. Improve test 2 (the only thing left is verifying bootstrap is successful based on node code that is not in the master).

### CODE REVIEW

File: tests/base_test_ci.py [hunk 1]
```diff
@@ -21,73 +28,47 @@ def setUp(self):
...
-        self.message = message.data
+        if self.phase != message.attributes['phase']:
+            return
+        logging.info(message)
+        self.messages.append(message.data.decode("utf-8"))
         message.ack()
-        self.endFlag = True
 
-    def send(self, data):
+    def create_phase(self, test, phase):
```
Comment (gavraz) [inline tests/base_test_ci.py]: Maybe just use the phase number as the identifier of the message.
Please note that it requires that each test that is run starts from scratch (i.e independent tests). It is also important for the trusty of the tests.

File: tests/base_test_ci.py [hunk 2]
```diff
@@ -21,73 +28,47 @@ def setUp(self):
...
+        self.phase = 'test_' + str(test) + '_phase_' + str(phase) + '_' + str(calendar.timegm(time.gmtime()))
+        return self.phase
+
+    def send(self, data, phase=None):
```
Comment (gavraz) [inline tests/base_test_ci.py]: Consider removing phase from arg and use self.phase instead.

File: tests/base_test_agent.py [hunk 3]
```diff
@@ -16,61 +19,69 @@ def __init__(self):
         self.topic_path_upstream = self.publisher_upstream.topic_path(project, topic_name_upstream)
 
         self.node = os.environ['NODE']
+        self.docker = Docker()
+        self.docker.stop('node_' + self.node)
+        logging.info('seeders:' + os.environ['SEEDERS'])
+        self.modify_seeders(os.environ['SEEDERS'])
+        self.phase = os.environ['PHASE']
+        self.node_port = config.CONFIG['node_port']
+        self.docker.start('docker run --network=devnet --name node_' + self.node + ' -p ' + str(self.node_port + int(self.node)) + ':' + str(self.node_port) + ' -v /root/spacemesh/devnet/logs' + self.node + ':/root/.spacemesh/nodes/ -v /root/spacemesh/devnet/cnf' + self.node + '/test.config.toml:/go/test.config.toml spacemesh/node:latest /go/src/github.com/spacemeshos/go-spacemesh/go-spacemesh -config /go/test.config.toml')
```
[the remark this answers is not in the document]
Comment (mosheka) [reply tests/base_test_agent.py]: Done

File: tests/base_test_agent.py [hunk 4]
```diff
@@ -16,61 +19,69 @@ def __init__(self):
...
         subscription_name_downstream = os.environ['SUBSCRIPTION_NAME_DOWNSTREAM']
         self.subscriber_downstream = pubsub_v1.SubscriberClient()
         self.subscription_path_downstream = self.subscriber_downstream.subscription_path(project, subscription_name_downstream)
+        time.sleep(5)
```
[the remark this answers is not in the document]
Comment (mosheka) [reply tests/base_test_agent.py]: Done

File: tests/base_test_agent.py [hunk 5]
```diff
@@ -16,61 +19,69 @@ def __init__(self):
...
     def send(self, data):
+        logging.info("sent: " + data)
         data = data.encode('utf-8')
-        self.publisher_upstream.publish(self.topic_path_upstream, data=data)
+        self.publisher_upstream.publish(self.topic_path_upstream, data=data, phase=self.phase)
 
     def get_node_id(self):
-        time.sleep(15)
-        node_id = 'NULL'
+        time.sleep(10)
```
[the remark this answers is not in the document]
Comment (mosheka) [reply tests/base_test_agent.py]: Done

**Pull request merged**